fuel-relayer/src/finalization_queue.rs

```diff
@@ -0,0 +1,521 @@
+
+    /// propagate new fuel block to pending_blocks
+    pub fn handle_fuel_block(&mut self, block: &Arc<SealedFuelBlock>) {
+        self.blocks.set_chain_height(block.header.height)
+    }
+
+    /// propagate new created fuel block to pending_blocks
+    pub async fn handle_created_fuel_block<P>(
+        &mut self,
+        block: &Arc<SealedFuelBlock>,
+        db: &mut dyn RelayerDb,
+        provider: &Arc<P>,
+    ) where
+        P: Middleware + 'static,
+    {
+        self.blocks.commit(block.header.height, db, provider).await;
+    }
+
+    pub async fn append_eth_logs(&mut self, logs: Vec<Log>) {
+        for log in logs {
+            self.append_eth_log(log).await;
+        }
+    }
+
+    /// Handle eth log events
+    pub async fn append_eth_log(&mut self, log: Log) {
+        let event = EthEventLog::try_from(&log);
+        if let Err(err) = event {
+            warn!(target:"relayer", "Eth Event not formated properly:{}",err);
+            return;
+        }
+        if log.block_number.is_none() {
+            error!(target:"relayer", "Block number not found in eth log");
+            return;
+        }
+        let removed = log.removed.unwrap_or(false);
+        let da_height = log.block_number.unwrap().as_u64() as DaBlockHeight;
+        let event = event.unwrap();
+        debug!("append inbound log:{:?}", event);
+        // bundle removed events and return
+        if removed {
+            self.bundle_removed_events(event, da_height);
+            return;
+        }
+        // apply all reverted event
+        if !self.bundled_removed_eth_events.is_empty() {
+            info!(
+                "Reorg happened on ethereum. Reverting {} logs",
+                self.bundled_removed_eth_events.len()
+            );
+
+            let mut lowest_removed_da_height = DaBlockHeight::MAX;
+
+            for (da_height, events) in
+                std::mem::take(&mut self.bundled_removed_eth_events).into_iter()
+            {
+                lowest_removed_da_height = DaBlockHeight::min(lowest_removed_da_height, da_height);
+                // mark all removed pending block commits as reverted.
+                for event in events {
+                    if let EthEventLog::FuelBlockCommited { block_root, height } = event {
+                        self.blocks
+                            .handle_block_commit(block_root, height.into(), da_height, true);
+                    }
+                }
+            }
+            // remove all blocks that were reverted. In best case those blocks heights and events are going
+            // to be reinserted in append eth events.
+            self.pending
+                .retain(|diff| diff.da_height < lowest_removed_da_height);
+        }
+        // apply new event to pending queue
+        self.append_da_events(event, da_height).await;
+    }
+
+    /// At begining we will ignore all event until event for new fuel block commit commes
+    /// after that syncronization can start.
```

Comment:
We should be able to track the finalized state of the contract with or without waiting and relying on blocks being committed to L1. For example, we may slow down L1 block commitments if the chain is very expensive or congested, but that shouldn't prevent or limit deposits from being processed to L2.

Reply:
That sounds right, but is this comment referring only to the initial sync, or consensus sync (i.e. already at the tip)?

Reply:
The comment is wrong, it should be updated. As you said Brandon, the state is finalized without relying on the block being committer.